packages/expo-build-properties/src/android.ts

```diff
@@ -53,10 +59,53 @@ export const withAndroidBuildProperties = createBuildGradlePropsConfigPlugin<Plu
       propName: 'android.enableProguardInReleaseBuilds',
       propValueGetter: (config) => config.android?.enableProguardInReleaseBuilds?.toString(),
     },
+    {
+      propName: 'android.flipper',
+      propValueGetter: (config) =>
+        typeof config.android?.flipper === 'string' ? config.android.flipper : undefined,
+    },
   ],
   'withAndroidBuildProperties'
 );
 
+export const withAndroidFlipper: ConfigPlugin<PluginConfigType> = (config, props) => {
+  const ANDROID_FLIPPER_KEY = 'FLIPPER_VERSION';
+  const FLIPPER_FALLBACK = '0.125.0';
+
+  // when not set, or set to enabled, make no changes
+  if (props.android?.flipper === undefined || props.android.flipper === true) {
+    return config;
+  }
+
+  return withGradleProperties(config, (c) => {
+    // check for flipper version in package. If set, use that
+    let existing: string | undefined;
+
+    const found = c.modResults.filter(
+      (item) => item.type === 'property' && item.key === ANDROID_FLIPPER_KEY
+    )?.[0];
+    if (found && found.type === 'property') {
+      existing = found.value;
+    }
+
+    // strip flipper key and re-add based on setting
+    c.modResults = c.modResults.filter(
+      (item) => !(item.type === 'property' && item.key === ANDROID_FLIPPER_KEY)
+    );
+
+    // if disabled, do not re-add
+    if (props.android?.flipper !== false) {
```

Comment:
what will happen if we don't add `FLIPPER_VERSION` property? should we also add in template?

```gradle
def flipperVersion = findProperty('FLIPPER_VERSION')
if (flipperVersion) {
    debugImplementation("com.facebook.flipper:flipper:${flipperVersion}")
    debugImplementation("com.facebook.flipper:flipper-network-plugin:${flipperVersion}") {
        exclude group:'com.squareup.okhttp3', module:'okhttp'
    }
    debugImplementation("com.facebook.flipper:flipper-fresco-plugin:${flipperVersion}")
}
```

Reply:
Taking another look through how Flipper's integrated, I don't think there's a clean way to "disable" flipper on android. I'll be removing the `false` option specifically for Android.

`templates/expo-template-bare-minimum/android/app/src/debug/java/com/helloworld/ReactNativeFlipper.java` is included in the build template and removing it would increase the surface area of the plugin to a level I don't think anyone's comfortable maintaining.